### ISSUE TITLE
Nvda does not announce checkbox status in software c cleaner

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#6887]
Nvda does not announce the state of the checkboxes when browsing the list of items to be deleted on the computer.
To test:
Open software c cleaner in its last version 5.27.
Press enter on the "clean" option.
Navigate with a tab to the tree.
Now navigate with the arrows up and down.
The expected is that the nvda read whether the option is selaged or not.
What happens is that nvda does not read the status of the option.
Press Space. It is expected that nvda will read if the option has been cleared or checked.
What happens is that the status of the selection is not informed.
Also when browsing this list it would be interesting to hear the information of the space of the files found, and the amount of files that will be deleted.
Nvda verção next 13881.
Follow the log.

IO - inputCore.InputManager.executeGesture (10:56:00):
Input: kb(laptop):space
IO - speech._speakSpellingGen (10:56:00):
Speaking character u'espa\xe7o'
IO - inputCore.InputManager.executeGesture (10:56:03):
Input: kb(laptop):space
IO - speech._speakSpellingGen (10:56:03):
Speaking character u'espa\xe7o'
IO - inputCore.InputManager.executeGesture (10:56:07):
Input: kb(laptop):NVDA+f1
INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (10:56:07):
Developer info for navigator object:
name: u'ccsetup527.exe'
role: ROLE_LISTITEM
states: STATE_FOCUSABLE, STATE_SELECTABLE, STATE_FOCUSED, STATE_SELECTED
isFocusable: True
hasFocus: 1
Python object: <NVDAObjects.Dynamic_UIItemListItemUIA object at 0x0ABA0B70>
Python class mro: (<class 'NVDAObjects.Dynamic_UIItemListItemUIA'>, <class 'NVDAObjects.UIA.UIItem'>, <class 'NVDAObjects.UIA.ListItem'>, <class 'NVDAObjects.UIA.UIA'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: ''
location: (286, 267, 574, 22)
value: ''
appModule: <'explorer' (appName u'explorer', process ID 24708) at address 5335a50>
appModule.productName: u'Sistema operacional Microsoft\xae Windows\xae'
appModule.productVersion: u'10.0.14393.0'
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
windowHandle: 199236
windowClassName: u'DirectUIHWND'
windowControlID: 0
windowStyle: 1442840576
windowThreadID: 53964
windowText: u''
displayText: u'ccsetup527.exe  14/ 02/ 2017   10:43 Aplicativo 9.045 KB'
UIAElement: <POINTER(IUIAutomationElement) ptr=0x78e5dc8 at ab5cc10>
UIA automationID: 2
UIA frameworkID: DirectUI
UIA runtimeID: (24708, 169425584, 0)
UIA providerDescription: [pid:24708,hwnd:0x0 Main(parent link):Unidentified Provider (unmanaged:explorerframe.dll)]
UIA className: UIItem

**Comment by fernando-jose-silva** [comment]
Using windows 10 14393 64bit

**Comment by Brian1Gaff** [comment]
Is this in an installed version of nvda. Ccleaner is one of th pieces of 
software that has issues in 10 with a portable version of nvda.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
----- Original Message -----

**Comment by fernando-jose-silva** [comment]
Yes, I did the tests on an installed nvda.
Thank you.

**Comment by bhavyashah** [comment]
@fernando-jose-silva Does this still occur for you? Using NVDA 2017.3 RC 1 and a recent version of CCleaner, Internet Explorer is the first item of the tree view and NVDA does report that it is expanded and checked. Pressing Spacebar also triggered NVDA to report the state change accurately. I do vaguely recall encountering the issue you describe in the past, but at present, probably due over time and versions, this has evidently been fixed. If you can confirm my results, we can safely close this ticket as worksforme. @ehollig

**Comment by fernando-jose-silva** [comment]
Unfortunately for me the problem still occurs.
I am using the nvda next 14327 and the "c cleaner" 5.33 64bits, the windows is the 10 15063.540 64bits.
I believe this should be dependent on windows and if the software is 64 bit.
Thank you very much.

**Comment by LivingForGod1991** [comment]
Hi. This is also the case for me. I’m also using NVDA Next snapshot 14327, CCleaner 5.33, and Windows 10 Home build 16257 64-bit.

**Comment by Brian1Gaff** [comment]
I have just tested this in windows 7. It works on the first item, but 
subsequent items only respond if you change their state, so to be sure its 
set as what you want you need to hit space twice, once to change it the 
other way then again to get it back as it was. Simply going up and down does 
not reveal the presence of checkboxes at all.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.

**Comment by fernando-jose-silva** [comment]
For me nvda never announces the status of checkboxes, regardless of their status.

Enviado do meu iPhone

**Comment by Brian1Gaff** [comment]
I think what I was getting at is that often you won't even know a checkbox 
exists in this part of ccleaner without toggling them.

Its only then that you heare they are there and the state its toggled to.
This particular part is also flakey as it uses uia I think and hence won't 
work on 7 and up on portable versions of nvda very reliably, escpecially 
after a scan.

Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.

**Comment by fernando-jose-silva** [comment]
If they could evaluate in windows 10, 64 bits, in my case nvda never announces the statos of these boxes

Enviado do meu iPhone

> Em 21 de ago de 2017, às 09:41, Brian Gaff <notifications@github.com> escreveu:
> 
> I think what I was getting at is that often you won't even know a checkbox 
> exists in this part of ccleaner without toggling them.
> 
> Its only then that you heare they are there and the state its toggled to.
> This particular part is also flakey as it uses uia I think and hence won't 
> work on 7 and up on portable versions of nvda very reliably, escpecially 
> after a scan.
> 
> Brian

**Comment by LivingForGod1991** [comment]
It’s that way in my case as well. NVDA never announces the checkboxes, even if I push the spacebar to toggle them. The first one is not even announced.

**Comment by Brian1Gaff** [comment]
It is indeed strange that although this effect is there for everyone, that 
some get it if you actually operate the invisible checkbox and some never 
see it at all.
 Would I perhaps be right in thinking that the ones who never see it at all 
are using windows 10?
 I'm using 7.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.

**Comment by k-kolev1985** [comment]
Yes, I also have this issue and for me NVDA does not see the check boxes as such - only as tree view items. I'm running the latest "next" snapshot of NVDA on a Windows 10 64-bit system. CCleaner is the latest version and it is also 64-bit, because that is what you get when you install CCleaner on a 64-bit system. But I think this is a problem of CCleaner and not of NVDA. I've been complaining to the CCleaner developers in their forum and on their Facebook page about accessibility issues related to high-contrast mode of Windows not supported by CCleaner, they say they'll look into it, but haven't done anything about it yet. So good luck make them listen to your complains... :)

**Comment by Brian1Gaff** [comment]
I only have the free version but yes you do have to use the 64 bit version. 
are you aware there is a portable archive version as well. I tend to use 
that a lot.
 However in the treeview of all of them I only get told there is a checkbox 
when I change it with space, or encounter it just after I altered it without 
moving focus.
 They also have a bug for xp users that means that it can only run on Intel 
processors. It really should exit cleanly with an error but it just falls 
over as sse2 is needed on the processor.
 So although we may complain about accessibility, it has had some other 
issues for years, and at present its still pretty usable in my opinion.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal email to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.

**Comment by Qchristensen** [comment]
Has anyone looked at this issue recently?  I had a report from a user on Twitter just now that this issue is still present.

**Comment by fernando-jose-silva** [comment]
I still have this problem.
windows 10 64bit 17763.348
nvda alfa 16874
c cleaner 5.54 64bit
log:
[cc.txt](https://github.com/nvaccess/nvda/files/2933728/cc.txt)

**Comment by agtheblade** [comment]
Hello. This is still true with 2019.3 beta 2 and ccleaner version 5.63.

**Comment by Brian1Gaff** [comment]
Unfortunately recent versions of ccleaner seem to have more serious issues 
where whole screens appear to be completely blank, so I do question thir 
ability to create accessible code. The other worry is their tendency to hide 
avast timed later installs in their installed versions of software.
 Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message -----

**Comment by Adriani90** [comment]
This works as expected in NVDA Version: alpha-21196,4840e5fb and CCleaner Version 5.72.7994. Closing as works for me.
